addons/mail/tests/test_mail_tools.py

```diff
@@ -107,6 +113,12 @@ def test_find_partner_from_emails(self):
             found = Partner._mail_find_partner_from_emails(['alfred_astaire@test.example.com'])
             self.assertEqual(found, [self.env['res.partner']])
 
+        # test users with same email, priority given to current user
+        # --------------------------------------------------------------
+        self.test_user.sudo().write({'email': '"Alfred Astaire" <%s>' % self.env.user.partner_id.email_normalized})
```

Comment:
Actually you could use self.user_admin that already exists, allowing to skip creation of a new user